### DISCUSSION TITLE
how to check the .net core runtime version in wix v4

### DISCUSSION THREAD

**Discussion by radz1** [wixtoolset/issues#6684] [Questions] [answered]
Hi, I am using the below code to check the core runtime using wix toolset v3. I am using it in bootstrapper bundle file. The problem with this code is, we need to add new key whenever the .net runtime core got released with new version. 

Can we have any way to use the regex or * like filter so that we can check if minimum version is 3.1.4 or later version? 

Please suggest any better way to verify the core runtime version with documentation, example. 

```
<Variable Name="NetCoreKey" Value="SOFTWARE\WOW6432Node\dotnet\Setup\InstalledVersions\x64\sharedfx\Microsoft.NETCore.App" Type="string"/>
        
<util:RegistrySearch Root="HKLM" Key="[NetCoreKey]" Result="value" Variable="NetCore3_x64_Installed_314" Win64="no" Format="raw" Value="3.1.4"/>

<bal:Condition Message="This application requires .NET Core Runtime 3.1.4 or later. ">
          <![CDATA[Installed OR 
          (NetCore3_x64_Installed_314 = "1" OR 
           NetCore3_x64_Installed_315 = "1" )]]>
    </bal:Condition>
```

**Comment by robmen** [comment, marked as answer]
Detecting .NET Core is pretty challenging. There is [this open issue](https://github.com/wixtoolset/issues/issues/6264) that some MSFT team members were working on for a while but not sure what the status of it is now.

You might follow up with the people there to see how you can help get the feature completed.
